include/inviwo/core/network/workspaceutils.h

```diff
@@ -31,19 +31,25 @@
 
 #include <inviwo/core/common/inviwocoredefine.h>
 
-#include <string>
+#include <string_view>
+#include <functional>
 
 namespace inviwo {
 
 class InviwoApplication;
 
 namespace util {
 
-void IVW_CORE_API updateWorkspaces(InviwoApplication* app);
+IVW_CORE_API void forEachWorkspaceInDirRecusive(std::string_view path,
+                                                std::function<void(std::string_view)> callback);
 
-void IVW_CORE_API updateRegressionWorkspaces(InviwoApplication* app);
+enum class DryRun { Yes, No };
 
-void IVW_CORE_API updateWorkspaces(InviwoApplication* app, const std::string& path);
+IVW_CORE_API void updateWorkspaces(InviwoApplication* app, std::string_view path, DryRun dryRun);
```

Comment:
Explain what DryRun means in this context.
I guess it just opens the workspaces, updates but does not save them?

Reply:
Probably also want to mention that update refers to the workspace version.
I know that this is more or less updating existing code but we could mention what happens if there are errors while loading a workspace. Is it still updated and saved?